src/utils/writeSensOutput.m

```diff
@@ -0,0 +1,146 @@
+                delay = cell(ITER,1);
+                gain = cell(ITER,1);
+                phase = cell(ITER,1);
+                aodEl =  cell(ITER,1);
+                aodAz= cell(ITER,1);
+                aoaAz= cell(ITER,1);
+                aoaEl= cell(ITER,1);
+                
+                for t = 1:ITER
+                    delaySensT = delaySens{t};
+                    delayCommT = delayComm{t};
+                    
+                    [delayT, order] = sort([delayCommT; delaySensT], 'ascend');
+                    delay{t} = delayT;
+                    gain{t}  = mergeSort(gainComm{t},gainSens{t},order);
+                    phase{t}  = mergeSort(phaseComm{t},phaseSens{t},order);
+                    aodEl{t}  = mergeSort(aodElComm{t},aodElSens{t},order);
+                    aodAz{t}  = mergeSort(aodAzComm{t},aodAzSens{t},order);
+                    aoaEl{t}  = mergeSort(aoaElComm{t},aoaElSens{t},order);
+                    aoaAz{t}  = mergeSort(aoaAzComm{t},aoaAzSens{t},order);
+                    
+                    
+                end
+                
+                s = struct('tx', tx-1, 'rx', rx-1,...
+                    'paaTx', txPaa-1, 'paaRx', rxPaa-1);
+                s.delay = delay;%mat2cell(single(sisoChSens(:,8)), rowDistSens);
+                s.gain  = gain;%mat2cell(single(real(sisoChSens(:,9))), rowDist);
+                s.phase = phase;%mat2cell(single(sisoChSens(:,18)), rowDist);
+                s.aodEl = aodEl;%mat2cell(single(sisoChSens(:,11)), rowDist);
+                s.aodAz = aodAz;%mat2cell(single(sisoChSens(:,10)), rowDist);
+                s.aoaEl = aoaEl;%mat2cell(single(sisoChSens(:,13)), rowDist);
+                s.aoaAz = aoaAz;%mat2cell(single(sisoChSens(:,12)), rowDist);
+                json = jsonencode(s);
+                str2remove =',null'; %Temporary string to remove
+                rem_ind_start = num2cell(strfind(json, str2remove)); % Find start string to remove
+                index2rm = cell2mat(cellfun(@(x) x:x+length(str2remove)-1,rem_ind_start,'UniformOutput',false)); % Create index of char to remove
+                json(index2rm) = [];
+                str2remove ='null';
+                rem_ind_start = num2cell(strfind(json, str2remove)); % Find start string to remove
+                index2rm = cell2mat(cellfun(@(x) x:x+length(str2remove)-1,rem_ind_start,'UniformOutput',false)); % Create index of char to remove
+                json(index2rm) = [];
+                fprintf(fid, '%s\n', json);
+            end
+        end
+    end
+end
+fclose(fid);
+
+
+end
+
+function x = getMat2D(x)
+x = permute(x,[1 3 2]);
+x = reshape(x,[size(x,1)*size(x,2) size(x,3)]);
+end
+
+
+function x = appendNan(x,n,m)
+if isempty(x)
+    x = nan(2,n,m);
+elseif size(x,3)<m
+    x(:, :, size(x,3)+1:m) = nan;
+    x(end+1,:,:) = nan;
+else
+    x(end+1,:,:) = nan;
+end
+end
+
+function x = mergeSort(y1,y2,order)
```

Comment:
i would call it mergeAndSort to avoid confusion with the famous sorting algorithm